api/envoy/service/ext_proc/v3alpha/external_processor.proto

```diff
@@ -182,99 +187,111 @@ message HttpTrailers {
 }
 
-// This message must be sent in response to a request_trailers
-// or response_trailers message.
+// This message must be sent in response to an HttpTrailers message.
 message TrailersResponse {
   // Instructions on how to manipulate the trailers
   HeaderMutation header_mutation = 1;
 }
 
-// TODO this is now the same as HeaderResponse. However, do we think
-// that will always be the case? Keeping it as a separate message
-// for now.
-// [#next-free-field: 6]
+// This message must be sent in response to an HttpBody message.
 message BodyResponse {
-  // If set, manipulate HTTP headers along with the body. Note that
-  // this is only possible if the headers have not been sent down
-  // the filter chain yet. Practically, then, this field will be ignored
-  // unless the request or response BodySendMode is BUFFERED or
-  // BUFFERED_PARTIAL, which means that Envoy is buffering the body
-  // in memory while waiting for this response.
-  HeaderMutation header_mutation = 1;
+  CommonResponse response = 1;
+}
 
-  // Request that the last body chunk delivered be modified.
-  BodyMutation body_mutation = 2;
-
-  // If set, add trailers to the message. If and only if this
-  // field is combined with the end_stream flag, then
-  // the trailers will be sent along with the request or
-  // response -- otherwise, this field is ignored.
-  config.core.v3.HeaderMap trailers = 3;
-
-  // If true, then replace the request and response with the contents
-  // of this message. If header_mutation is set and headers have not
-  // already been forwarded down the filter chain, apply it to the
-  // headers. If body_mutation is set and contains a body, then add that
-  // body to the request or response, even if one does not already exist --
-  // otherwise, clear the body. Any additional body and trailers
-  // received from downstream will be ignored, and any additional messages
-  // from the remote server on either the request or response path will
-  // also be ignored for the duration of this request or response.
-  bool end_of_stream = 4;
+// This message contains common fields between header and body responses.
+// [#next-free-field: 6]
+message CommonResponse {
+  enum ResponseStatus {
+    // Apply the mutation instructions in this message to the
+    // request or response, and then continue processing the filter
+    // stream as normal. This is the default.
+    CONTINUE = 0;
+
+    // Replace the request or response with the contents
+    // of this message. If header_mutation is set, apply it to the
+    // headers. If body_mutation is set and contains a body, then add that
+    // body to the request or response, even if one does not already exist --
+    // otherwise, clear the body. Any additional body and trailers
+    // received from downstream will be ignored, and any additional messages
+    // from the remote server on either the request or response path will
+    // also be ignored for the duration of this request or response.
```

Comment:
I think this text is wrong? Isn't the point of this that you want the remove server to be able to send more data, but have it be streamed?

Reply:
I was trying to distinguish this use case from the "pause and do not end stream" status. But I'd rather just take that status out and simplify things a bit.

Reply:
I'm not sure I follow. With how this is written I'm not sure what would ever continue/finish the stream?

Reply:
I can see how that reads confusingly and I'm going to take it out. I think it's clear enough without it.